CMakeLists.txt

```diff
@@ -98,6 +92,13 @@ project(box86 C ASM)
 
 enable_testing()
 
+string(COMPARE EQUAL "${CMAKE_SYSTEM_PROCESSOR}" "i686"  _x86)
+string(COMPARE EQUAL "${CMAKE_SYSTEM_PROCESSOR}" "x86_64"  _x86_64)
```

Comment:
The test on line 787 https://github.com/ptitSeb/box86/pull/1008/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR787 expect those variable to be about host. So maybe a host versio of those are needed too.

Reply:
imo, I don't think there's a need to prevent `make install`.
For example, we could do `make DESTDIR="$PWD/package" install`, installing all files to a safe place for my own packaging purpose.

And there's no issue with box64:
https://github.com/ptitSeb/box64/blob/70a6d9674876c4622b42b8913a9ee910a9e65c12/CMakeLists.txt#L1124-L1127

I modified this to be consistent with box64.

Reply:
Ok, fair enough.